shared/studio/components/explainVis/codeEditorContexts.tsx

```diff
@@ -92,9 +90,87 @@ export const CodeEditorExplainContexts = observer(function ExplainContexts({
       }
 
       setCtxRects(rects);
+    }, []);
+
+    useImperativeHandle<unknown, ExplainHighlightsRef>(
+      ref,
+      () => ({updateContextRects}),
+      [updateContextRects]
+    );
+
+    const [_, theme] = useTheme();
+    const palette = theme === Theme.light ? lightPalette : darkPalette;
+    const location = useLocation();
+    const isEditor = location.pathname.includes("editor");
```

Comment:
Can `isEditor` just be a prop that gets passed to the component.

Reply:
oh yes, ofc